quiche/src/lib.rs

```diff
@@ -5548,12 +6816,20 @@ fn drop_pkt_on_err(
     Error::Done
 }
 
+fn addr_tuple_to_string(from: SocketAddr, to: SocketAddr) -> String {
```

Comment:
I think this avoids the intermediate allocation:
```
use std::net::SocketAddr;

fn main() {
    println!("Hello {}", AddrTupleFmt("1.2.3.4:80".parse().unwrap(), "4.5.6.7:80".parse().unwrap()));
}

struct AddrTupleFmt(SocketAddr, SocketAddr);

impl std::fmt::Display for AddrTupleFmt {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        let AddrTupleFmt(from, to) = &self;
        if from.ip().is_unspecified() || to.ip().is_unspecified() {
            Ok(())
        } else {
            f.write_fmt(format_args!("({} -> {})", from, to))
        }
    }
}
```

Reply:
Good idea, thanks.